returnn/tf/layers/rec.py

```diff
@@ -1156,6 +1156,12 @@ def add_templated_layer(lself, name, layer_class, **layer_desc):
         if layer_ in ConstructCtx.partially_finished:
           if lself.got_uninitialized_deps_count == 0:  # in this case, we safely know that it is finished
             ConstructCtx.partially_finished.remove(layer_)
+            # Sub-layers were also added to 'partially_finished' to re-init them as well if necessary.
+            # But 'add_templated_layer' is not called for sub-layers, so clean them up here as well.
+            for sub_layer in layer_.sub_layers.values():
+              if sub_layer in ConstructCtx.partially_finished:
+                ConstructCtx.partially_finished.remove(sub_layer)
```

Comment:
@albertz This is not needed for the test to succeed. Is it needed at all? It felt like it. Also, it might be not the right place to remove the sub-layers. Maybe in `__call__()`? This template construction is still to complex for me 😆

Reply:
I don't know... The whole logic about sublayers seems somewhat complex. But without looking at it more deeply, I don't really see a good solution now.

Reply:
Yes, sublayers are complicated to handle. But also because `_construct_template()` is already very complex...